metrics/firebase/test/firestore/rules/project-groups-test.js

```diff
@@ -1,237 +1,325 @@
+
+    await assertFails(
+      passwordProviderAllowedDomainApp.collection(collectionName).doc("2").update({
+        name: "name",
+        projectIds: testProjectIds,
+      })
+    );
+  });
+
+  it("does not allow creating a project group with name size greater than 255", async () => {
+    const testName = "a".repeat(256);
+
+    await assertFails(
+      passwordProviderAllowedDomainApp.collection(collectionName).add({
+        name: testName,
+        projectIds: [],
+      })
+    );
+  });
+
+  it("does not allow creating a project group with project ids length greater than 20", async () => {
+    const testProjectIds = [...Array(21)].map((_, i) => `${i}`);
+
+    await assertFails(
+      passwordProviderAllowedDomainApp.collection(collectionName).add({
+        name: "name",
+        projectIds: testProjectIds,
+      })
+    );
+  });
+
+  /**
+   * The google auth provider user specific tests
+   */
+
+  it("does not allow to create a project group by an authenticated google user with not allowed email domain", async () => {
+    await assertFails(
+      googleProviderNotAllowedDomainApp
+        .collection(collectionName)
+        .add(getProjectGroup())
+    );
+  });
+
+  it("allows creating a project group by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp.collection(collectionName).add(getProjectGroup())
+    );
+  });
+
+  it("does not allow to read project groups by an authenticated google user with not allowed email domain", async () => {
+    await assertFails(
+      googleProviderNotAllowedDomainApp.collection(collectionName).get()
+    );
+  });
+
+  it("allows reading a project group by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp.collection(collectionName).get()
+    );
+  });
+
+  it("does not allow to update a project group by an authenticated google user with not allowed email domain", async () => {
+    await assertFails(
+      googleProviderNotAllowedDomainApp
+        .collection(collectionName)
+        .doc("2")
+        .update(getProjectGroup())
+    );
+  });
+
+  it("allows updating a project group by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp
+        .collection(collectionName)
+        .doc("2")
+        .update(getProjectGroup())
+    );
+  });
+
+  it("does not allow to delete a project group by an authenticated google user with not allowed email domain", async () => {
+    await assertFails(
+      googleProviderNotAllowedDomainApp.collection(collectionName).doc("1").delete()
+    );
+  });
+
+  it("allows deleting a project group by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp.collection(collectionName).doc("1").delete()
+    );
+  });
```

Comment:
Let's arrange tests in the order it is in the builds-test.js:

PasswordProviderAllowed
Create
Read
Update
Delete

PasswordProviderNowAllowed
CRUD (same as above)

GoogleProviderAllowed
CRUD (same as above)

GoogleProviderNotAllowed
CRUD (same as above)

Unauthenticated
CRUD (same as above)